workspaces/api-gateway/src/api/admin/products/services/CreateProductService.ts

```diff
@@ -0,0 +1,47 @@
+import { Injectable, InternalServerErrorException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import type { Repository } from 'typeorm';
+
+import { createSlug } from 'utils';
+import { Product } from 'database/models/product.entity';
+import { CheckProductExistedService } from 'api/admin/products/services/CheckProductExistedService';
+import { CreateProductCategoryService } from 'api/admin/products/services/CreateProductCategoryService';
+import { GetCategoryByKeyService } from 'api/admin/categories/services/GetCategoryByKeyService';
+import type { CreateProductDto } from 'api/admin/products/products.dto';
+
+@Injectable()
+export class CreateProductService {
+  constructor(
+    @InjectRepository(Product)
+    private productRepository: Repository<Product>,
+    private checkProductExistedService: CheckProductExistedService,
+    private createProductCategoryService: CreateProductCategoryService,
+    private getCategoryByKeyService: GetCategoryByKeyService,
+  ) {}
+
+  public async exec({ categoryId, name }: CreateProductDto): Promise<Product> {
+    const checkCategory = await this.getCategoryByKeyService.exec(categoryId);
+
+    const newProduct = this.productRepository.create({
+      name: name,
+      category: checkCategory,
+      slug: await createSlug(
+        {
+          name: name,
+          columnName: 'slug',
+        },
+        this.checkProductExistedService.exec.bind(this),
```

Comment:
why bind `this` here? Try to remove the `private productRepository: Repository<Product>` in `checkProductExistedService` to see the result